Sources/cctl/cctl.swift

```diff
@@ -70,6 +75,19 @@ struct Application: AsyncParsableCommand {
     )
 }
 
+struct CommandContext {
+    let appRoot: URL
+    let eventLoopGroup: any EventLoopGroup
+    let httpClientProvider: HTTPClientProvider
+    let contentStore: ContentStore
+    let imageStore: ImageStore
+
+    func shutdown() throws {
```

Comment:
This approach seems very error prone, even for instances where I'm not pulling an image. It will be very easy to forget to call shutdown.

Reply:
Per our discussion, simplified to keep RegistryClient opinionated and just fix the proxy handling there and in ProxyUtils.